### PR TITLE
fix(provider/gce): Fix seqence generation for server groups.

### PR DESCRIPTION
**Pull request by jtk54**
Also fixes an NPE when applying monikers. @jervi PTAL.

### CODE REVIEW

**Comment by jervi** [comment]
Nice 👍

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy [hunk 1]
```diff
@@ -50,24 +50,24 @@ class GoogleLabeledResourceNamerSpec extends Specification {
...
+    "cass-nccpintegration-random-junk-d0prod-z0useast1a-v003" | [:]                                   || "cass"       | "cass-nccpintegration-random-junk-d0prod-z0useast1a" | "nccpintegration" | "random-junk" | 3
+    "cass-nccpintegration-random-junk-d0prod-z0useast1a-v003" | [(APP): "myApp"]                      || "myApp"      | "cass-nccpintegration-random-junk-d0prod-z0useast1a" | "nccpintegration" | "random-junk" | 3
+    "cass-nccpintegration-random-junk-v003"                   | [(CLUSTER): "myCluster"]              || "cass"       | "myCluster"                                          | "nccpintegration" | "random-junk" | 3
+    "cass-nccpintegration-random-junk-v003"                   | [(STACK): "myStack"]                  || "cass"       | "cass-myStack"                                       | "myStack"         | "random-junk" | 3
+    "cass-nccpintegration-random-junk-v003"                   | [(STACK): "myStack", (DETAIL): ""]    || "cass"       | "cass-myStack"                                       | "myStack"         | ""            | 3
```
Comment (jtk54) [inline clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy]: What is the particular use tested in this case? Is it OK to have empty string detail -> empty string expected detail @jervi ?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java [hunk 2]
```diff
@@ -99,9 +99,6 @@ private static String getClusterName(String app, String stack, String detail) {
 
   private static <T> void setIfPresent(Consumer<T> setter, T value) {
     if (value != null) {
-      if (value.equals("")) {
-        value = null;
-      }
```
Comment (jervi) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java]: It is actually this way on purpose. The use case is if you have some legacy infrastructure that is named in a way that Frigga parses as something, you can overwrite that part by setting an empty label. Without this feature, you can only overwrite it with another value, not remove/ignore it. E.g. if you have a legacy instance group named `myApp-foo-bar` and you want it to appear in Spinnaker as `myApp-foo`, you can do that by setting an empty `spinnaker-moniker-detail` label.

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy (hunk 1 above)
Comment (jervi) [reply clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy]: See my comment above

Comment (jervi) [reply clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy]: But I guess it's OK. The result will be the same, I think.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java (hunk 2 above)
Comment (jtk54) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java]: I think you could omit/delete the `details` moniker label (while including the others) to remove the detail from the legacy server group.

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy (hunk 1 above)
Comment (jtk54) [reply clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamerSpec.groovy]: :+1:

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java (hunk 2 above)
Comment (jervi) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java]: Yeah. I think this block was something I should have removed earlier anyway and never should have ended up in the PR. Thanks for fixing :)

**Pull request merged**